step.go

```diff
@@ -7,133 +7,130 @@ import (
 	"net/url"
 	"os"
 	"strings"
+
+	"github.com/bitrise-io/go-utils/log"
 )
 
 // -----------------------
 // --- Constants
 // -----------------------
-
 const (
 	baseURL = "https://api.hipchat.com/v1"
 )
 
+// ConfigsModel ...
+type ConfigsModel struct {
+	//oAuth
+	token  string
+	roomID string
+
+	//onSuccess
+	fromName     string
+	message      string
+	messageColor string
+
+	//onFail
+	fromNameOnError     string
+	messageOnError      string
+	messageColorOnError string
+
+	//settings
+	messageFormat     string
+	isBuildFailedMode string
+}
+
 // -----------------------
 // --- Functions
 // -----------------------
 
-func logFail(format string, v ...interface{}) {
-	errorMsg := fmt.Sprintf(format, v...)
-	fmt.Printf("\x1b[31;1m%s\x1b[0m\n", errorMsg)
-	os.Exit(1)
-}
+func createConfigsModelFromEnvs() ConfigsModel {
+	return ConfigsModel{
+		token:  os.Getenv("auth_token"),
+		roomID: os.Getenv("room_id"),
 
-func logWarn(format string, v ...interface{}) {
-	errorMsg := fmt.Sprintf(format, v...)
-	fmt.Printf("\x1b[33;1m%s\x1b[0m\n", errorMsg)
-}
+		fromName:     os.Getenv("from_name"),
+		message:      os.Getenv("message"),
+		messageColor: os.Getenv("message_color"),
 
-func logInfo(format string, v ...interface{}) {
-	fmt.Println()
-	errorMsg := fmt.Sprintf(format, v...)
-	fmt.Printf("\x1b[34;1m%s\x1b[0m\n", errorMsg)
-}
+		fromNameOnError:     os.Getenv("from_name_on_error"),
+		messageOnError:      os.Getenv("message_on_error"),
+		messageColorOnError: os.Getenv("message_color_on_error"),
 
-func logDetails(format string, v ...interface{}) {
-	errorMsg := fmt.Sprintf(format, v...)
-	fmt.Printf("  %s\n", errorMsg)
+		messageFormat:     os.Getenv("message_format"),
+		isBuildFailedMode: os.Getenv("STEPLIB_BUILD_STATUS"),
```

Comment:
We should use `BITRISE_BUILD_STATUS` instead (http://devcenter.bitrise.io/faq/available-environment-variables/)